File: 2019/1_The_Tyranny_of_the_Rocket_Equation/part_1.ts
import { readFileSync } from 'fs'
import { floor, map, split, sum, toNumber } from 'lodash'

const input = split(readFileSync('input.txt', 'utf-8'), '\n')

console.log(sum(map(input, (number) => floor(toNumber(number) / 3) - 2)))

File: 2023/1_Trebuchet/part_1.rb
lines = File.read('input.txt').split("\n")

start_time = Time.now

def number(line, cursor, iterator)
  loop do
    char = line[cursor]
    return char if (1..9).include?(char.to_i)

    cursor += iterator
  end
end

numbers_for_lines =
  lines.map do |line|
    "#{number(line, 0, 1)}#{number(line, line.length - 1, -1)}".to_i
  end

end_time = Time.now

puts numbers_for_lines.sum
# => 54450

puts end_time - start_time
